Extension/src/LanguageServer/cppBuildTaskProvider.ts

```diff
@@ -380,22 +392,29 @@ class CustomBuildTaskTerminal implements Pseudoterminal {
 
     private async doBuild(): Promise<any> {
         // Do build.
-        let command: string = util.resolveVariables(this.command);
-        let activeCommand: string = command;
+        let resolvedCommand: string | util.IQuotedString | undefined;
+        if (util.isString(this.command)) {
+            resolvedCommand = util.resolveVariables(this.command);
+        } else {
+            resolvedCommand = {
+                value: util.resolveVariables(this.command.value),
+                quoting: this.command.quoting
+            };
+        }
 
         // Create the exe folder path if it doesn't exist.
         const exePath: string | undefined = util.resolveVariables(util.findExePathInArgs(this.args));
         util.createDirIfNotExistsSync(exePath);
 
         this.args.forEach((value, index) => {
-            value = util.quoteArgument(util.resolveVariables(value));
```

Comment:
If both uses of `util.quoteArgument` are removed in this PR and we don't need that function anymore, I think we should delete the implementation.